### PR TITLE
Add release 10.1.0

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
This adds the release entry for Apache NuttX 10.1.0 to the website.

## Testing
Locally built the site image

### CODE REVIEW

File: _releases/10.1.0.md [hunk 1]
```diff
@@ -0,0 +1,998 @@
...
+layout: page
+released: true
+apache: true
+title: 10.1.0
+date: 2021-5-26
+summary: >
+    Release v10.1.0
+
+artifact-root: "https://archive.apache.org/dist/incubator/nuttx/10.1.0"
+checksum-root: "https://archive.apache.org/dist/incubator/nuttx/10.1.0"
```
Comment (Ouss4) [inline _releases/10.1.0.md:11]:
```suggestion
artifact-root: "https://archive.apache.org/incubator/nuttx/10.1.0"
checksum-root: "https://archive.apache.org/incubator/nuttx/10.1.0"
```

**Comment by davids5** [comment]
> -This adds the release entry for Apache NuttX 10.0.0 to the website.
+This adds the release entry for Apache NuttX 10.1.0 to the website.

**Comment by Ouss4** [comment]
> > -This adds the release entry for Apache NuttX 10.0.0 to the website.
> > +This adds the release entry for Apache NuttX 10.1.0 to the website.

I've edited the description.

**Review by Ouss4 — APPROVED** [review]

**Comment by btashton** [comment]
Thanks @jerpelea we just need to wait 48hr after you uploaded the artifacts to merge this and send the email out.

**Comment by acassis** [comment]
Hi @btashton I think 48h already passed, may we merge it?

**Pull request merged**